From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sat, 16 Dec 2017 17:12:16 +0100
Subject: add debian to --with-distro switch

---
 configure.ac | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 6aa1aa9..6f14b46 100644
--- a/configure.ac
+++ b/configure.ac
@@ -138,7 +138,7 @@ AS_IF([test "$BUILDOS" = "linux"], [
 AC_MSG_CHECKING([specified Linux distribution name])
 AC_ARG_WITH([distro], AS_HELP_STRING([--with-distro], [set distribution name, used for sysconfig and initscripts only. available: rhel, suse or sles]),
   [with_distro=${withval}],[with_distro=rhel])
-if test "$with_distro" != "rhel" -a "$with_distro" != "suse" -a \
+if test "$with_distro" != "rhel" -a "$with_distro" != "suse" -a "$with_distro" != "debian" -a\
    "$with_distro" != "sles"; then
 	AC_MSG_ERROR([no initscript for $with_distro known])
 else
