From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 26 Apr 2015 15:15:01 +0200
Subject: Add include for database configuration

---
 installer/linux/server.cfg | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/installer/linux/server.cfg b/installer/linux/server.cfg
index 02792b9..0b72e69 100644
--- a/installer/linux/server.cfg
+++ b/installer/linux/server.cfg
@@ -475,3 +475,5 @@ local_admin_users = root kopano
 
 # Enable/disable reminders for shared stores
 #shared_reminders = yes
+
+!include debian-db.cfg
