From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Sun, 26 Apr 2015 15:15:01 +0200
Subject: Add include for database configuration

---
 installer/linux/server.cfg | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/installer/linux/server.cfg b/installer/linux/server.cfg
index fe19c70..dd6f366 100644
--- a/installer/linux/server.cfg
+++ b/installer/linux/server.cfg
@@ -479,3 +479,5 @@ embedded_attachment_limit = 20
 # is taken to be received via a proxy. An empty value disables proxy detection
 # and the value of '*' is used to indicate that all connections are proxied
 proxy_header = 
+
+!include debian-db.cfg
