From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 22 Feb 2018 00:30:43 +0100
Subject: build: guard against external version bumps (2)

---
 versionheader.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/versionheader.sh b/versionheader.sh
index 24e4c49..da21757 100755
--- a/versionheader.sh
+++ b/versionheader.sh
@@ -7,11 +7,12 @@ if test -z "$acver"; then
 	exit 1
 fi
 version=$(cat version 2>/dev/null)
-if test "${version:0:${#acver}}" != "$acver"; then
+if test -z "$version"; then version="$acver"; fi
+if perl -e 'exit!($ARGV[1]!~/^\Q$ARGV[0]\E\b/)' "$acver" "$version"; then
 	echo "You can only append, not override, the version specified in AC_INIT."
 	exit 1
 fi
-set -- $(sed -e 's/ .*//g;s/[^0-9a-z][^0-9a-z]*/ /g' <version 2>/dev/null)
+set -- $(echo "$version" | sed -e 's/ .*//g;s/[^0-9a-z][^0-9a-z]*/ /g' 2>/dev/null)
 major_version="$1"
 minor_version="$2"
 micro_version="$3"
