From: Jan Engelhardt <jengelh@inai.de>
Date: Thu, 7 Mar 2019 13:56:32 +0100
Subject: libserver: reduce indent in SOT::QueryRowByColumn

---
 provider/libserver/ECStoreObjectTable.cpp | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/provider/libserver/ECStoreObjectTable.cpp b/provider/libserver/ECStoreObjectTable.cpp
index e61a611..6f665e5 100644
--- a/provider/libserver/ECStoreObjectTable.cpp
+++ b/provider/libserver/ECStoreObjectTable.cpp
@@ -870,15 +870,16 @@ ECRESULT ECStoreObjectTable::QueryRowDataByColumn(ECGenericObjectTable *lpThis,
 	}
 
 	for (const auto &col : mapColumns)
-		for (const auto &ob : mapObjIds)
-			if (setDone.count({ob.second, col.second}) == 0) {
-				// We may be overwriting a value that was retrieved from the cache before.
-				if (soap == NULL && lpsRowSet->__ptr[ob.second].__ptr[col.second].ulPropTag != 0)
-					FreePropVal(&lpsRowSet->__ptr[ob.second].__ptr[col.second], false);
-				CopyEmptyCellToSOAPPropVal(soap, col.first, &lpsRowSet->__ptr[ob.second].__ptr[col.second]);
-				cache->SetCell(const_cast<sObjectTableKey *>(&ob.first),
-					col.first, &lpsRowSet->__ptr[ob.second].__ptr[col.second]);
-			}
+		for (const auto &ob : mapObjIds) {
+			if (setDone.count({ob.second, col.second}) != 0)
+				continue;
+			// We may be overwriting a value that was retrieved from the cache before.
+			if (soap == NULL && lpsRowSet->__ptr[ob.second].__ptr[col.second].ulPropTag != 0)
+				FreePropVal(&lpsRowSet->__ptr[ob.second].__ptr[col.second], false);
+			CopyEmptyCellToSOAPPropVal(soap, col.first, &lpsRowSet->__ptr[ob.second].__ptr[col.second]);
+			cache->SetCell(const_cast<sObjectTableKey *>(&ob.first),
+				col.first, &lpsRowSet->__ptr[ob.second].__ptr[col.second]);
+		}
 	return erSuccess;
 }
 
