From: Jan Engelhardt <jengelh@inai.de>
Date: Sun, 31 Mar 2019 23:34:34 +0200
Subject: server: recognize "TLSv1.0" for backward compat

---
 provider/server/ECSoapServerConnection.cpp | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/provider/server/ECSoapServerConnection.cpp b/provider/server/ECSoapServerConnection.cpp
index a026921..ff103bf 100644
--- a/provider/server/ECSoapServerConnection.cpp
+++ b/provider/server/ECSoapServerConnection.cpp
@@ -110,7 +110,8 @@ int kc_ssl_options(struct soap *soap, char *protos, const char *ciphers,
 		else if (strcasecmp(ssl_name, SSL_TXT_SSLV2) == 0)
 			ssl_proto = 0x01;
 #endif
-		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1) == 0)
+		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1) == 0 ||
+		    strcasecmp(ssl_name, "TLSv1.0") == 0)
 			ssl_proto = 0x04;
 #ifdef SSL_TXT_TLSV1_1
 		else if (strcasecmp(ssl_name, SSL_TXT_TLSV1_1) == 0)
