From: =?utf-8?q?Guido_G=C3=BCnther?= <agx@sigxcpu.org>
Date: Fri, 2 Nov 2018 16:19:17 +0100
Subject: Use proper python pkg config

---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index b1dfcb0..a5a61ab 100644
--- a/configure.ac
+++ b/configure.ac
@@ -671,11 +671,12 @@ AS_IF([test -n "$PYTHON"], [
 	PYTHON=$(echo "$PYTHON $@")
 	PYTHON_MAJOR_VERSION=$($PYTHON -c "from __future__ import print_function;import sys;print(sys.version_info[[0]])" 2>/dev/null)
 ])
+AS_IF([test "$PYTHON_MAJOR_VERSION" = "3"], [PYTHON_PC="python3"], [PYTHON_PC="python"])
 AH_TEMPLATE([ENABLE_PYTHON], [Build dagent/spooler with Python plugin support])
 AC_ARG_ENABLE([python], [AS_HELP_STRING([--enable-python], [Enable building of Python bindings (default: auto)])],
 	[want_python="$enableval"], [want_python=auto])
 AS_IF([test "$want_python" != no], [
-	PKG_CHECK_MODULES([PYTHON], [python], [], [:])
+	PKG_CHECK_MODULES([PYTHON], [$PYTHON_PC], [], [:])
 	AS_IF([test -n "$PYTHON_LIBS"], [
 		AC_PATH_PROG([SWIG_EXEC], [swig])
 		AM_PATH_PYTHON([2.5])
